src/lumigo_opentelemetry/external/botocore/extensions/sqs.py

```diff
@@ -77,34 +77,56 @@ class ContextableList(list):
         def __init__(self, messages, span: Span):
             super().__init__(messages)
             self._span = span
-            self._iteration_ongoing = False
+            self._scope_depth = 0
             self._token = None
 
+        # Iterator support
         def __getitem__(self, key: int) -> Any:
             try:
                 return super(_SqsExtension.ContextableList, self).__getitem__(key)
             finally:
-                # NonRecordingSpan is what we get when there is no span in the context
-                if self._iteration_ongoing and isinstance(get_current_span(), NonRecordingSpan) and self._span:
-                    # If there is not current active span (e.g. in case of an overarching entry span),
-                    # we restore as active the "ReceiveMessage" span. We neither record exceptions on it
-                    # nor set statuses, as the "ReceiveMessage" span is already ended when it is used as
-                    # parent by spans created while iterating on the messages.
-                    self._token = context_api.attach(context_api.set_value(_SPAN_KEY, self._span))
+                pass
+                # self._enter_scope()
 
         def __iter__(self) -> Generator:
             # Try / finally ensure that we detach the context over a `break` inside an iteration
             try:
-                # We use self._iteration_ongoing to guard against leaking span
-                # context over a direct index-based access like `response["Messages"][0]`
-                self._iteration_ongoing = True
+                self._enter_scope()
 
                 index = 0
                 while index < len(self):
                     yield self[index]
                     index = index + 1
             finally:
-                self._iteration_ongoing = False
+                self._exit_scope()
+
+        # Scope support
+        def __enter__(self):
+            self._enter_scope()
+
+        def __exit__(self, *args):
+            self._exit_scope()
+
+        def _enter_scope(self):
+            # NonRecordingSpan is what we get when there is no span in the context
+            if self._scope_depth == 0 and isinstance(get_current_span(), NonRecordingSpan) and self._span:
+                # If there is not current active span (e.g. in case of an overarching entry span),
+                # we restore as active the "ReceiveMessage" span. We neither record exceptions on it
+                # nor set statuses, as the "ReceiveMessage" span is already ended when it is used as
+                # parent by spans created while iterating on the messages.
+                self._token = context_api.attach(context_api.set_value(_SPAN_KEY, self._span))
+                self._scope_depth = 1
+            else:
+                self._scope_depth = self._scope_depth + 1
+
+        def _exit_scope(self):
+            if not self._scope_depth:
+                # Something very fishy going on!
```

Comment:
log it as an error

Reply:
This can be plenty spammy, too risky imo